### ISSUE TITLE
JSON Output has Logs for Standard Error

### ISSUE THREAD

**Issue by nhalstead** [tenable/terrascan#1378]
* terrascan version: v1.15.2 (via Docker Image Tag Latest)
* Operating System: Docker Container

### Description

The output type of JSON also contains errors that should be dumped to standard error.

### What I Did

```bash
terrascan scan --output json --log-type console --use-colors false > report.json 2> error.txt || true
```

#### error.txt

```
Unable to find image 'tenable/terrascan:latest' locally
latest: Pulling from tenable/terrascan
2408cc74d12b: Already exists
...
6eb609282c41: Pull complete
Digest: sha256:e36aed65df132ffb722b6b524f703eb819cbf17efb7ba5a9397a4b908e87e30e
Status: Downloaded newer image for tenable/terrascan:latest
2022-08-29T20:56:55.100Z	[31merror[0m	opa/engine.go:419	resource ID has an invalid format Dockerfile
```

#### report.json:

```json
ERROR: json: cannot unmarshal number into Go struct field Function_Environment.Properties.Environment.Variables of type string
{
  "results": {
    "scan_errors": [
      {
...
```

**Comment by nhalstead** [comment]
I was looking to add a [segment of code here](https://github.com/tenable/terrascan/blob/master/pkg/logging/logger.go#L82) to pipe the logs to `stderr` (as indicated from his [SO Post](https://stackoverflow.com/a/68476472/5779200)), but without knowing the full scope of the logger and how Tenable would like this to work, I couldn't make the change on my own.

---

Perhaps a command flag for the results json file to be dumped? The output would be strictly the results in the selected format from `--output json` so an additional flag like `--output-file result.json`.

**Comment by VeraBE** [comment]
I think it's very similar to this issue: https://github.com/tenable/terrascan/issues/1326

**Comment by nhalstead** [comment]
Hi @VeraBE, The issue is not about the error itself, it's the fact that the error has come through on an output rather than standard error

**Comment by VeraBE** [comment]
Yes, the other issues is also about that, and I believe I fixed it in the goformation library, Terrascan just needs to update that dependency
